OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/preferencecenter/OptimovePreferenceCenter.java

```diff
@@ -0,0 +1,299 @@
+import com.optimove.android.OptimoveConfig;
+import com.optimove.android.main.common.UserInfo;
+import com.optimove.android.main.tools.networking.HttpClient;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import okhttp3.Response;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class OptimovePreferenceCenter {
+    private static final String TAG = OptimovePreferenceCenter.class.getName();
+    private static OptimovePreferenceCenter shared;
+    private static OptimoveConfig config;
+
+    /**
+     * package
+     */
+    static ExecutorService executorService;
+    /**
+     * package
+     */
+    static final Handler handler = new Handler(Looper.getMainLooper());
+
+    public interface PreferencesGetHandler {
+        void run(@Nullable Preferences preferences);
+    }
+
+    public interface PreferencesSetHandler {
+        void run(Boolean result);
+    }
+
+    public enum Channel {
+        MOBILE_PUSH(489),
+        WEB_PUSH(490),
+        SMS(493);
+        private final int channel;
+
+        Channel(int channel) {
+            this.channel = channel;
+        }
+
+        @NonNull
+        public int getValue() {
+            return channel;
+        }
+
+        public static Channel getChannelByValue ( int value) {
+            switch (value) {
+                case 489:
+                    return MOBILE_PUSH;
+                case 490:
+                    return WEB_PUSH;
+                case 493:
+                    return SMS;
+                default:
+                    throw new IllegalArgumentException("Preference center does not support channel " + value);
+            }
+        }
+    }
+
+    public static OptimovePreferenceCenter getInstance() {
+        if (shared == null) {
+            throw new IllegalStateException("OptimovePreferenceCenter is not initialized");
+        }
+        return shared;
+    }
+
+    /**
+     * Initializes an instance of OptimovePreferenceCenter
+     *
+     * @param currentConfig current config
+     */
+    public static void initialize(OptimoveConfig currentConfig) {
+        shared = new OptimovePreferenceCenter();
+        config = currentConfig;
+
+        executorService = Executors.newSingleThreadExecutor();
+    }
+
+    /**
+     * Asynchronously runs preferences get handler on UI thread. Handler receives a single argument Preferences
+     *
+     * @param preferencesGetHandler handler
+     */
+    public void getPreferencesAsync(@NonNull PreferencesGetHandler preferencesGetHandler) {
+        UserInfo userInfo = Optimove.getInstance().getUserInfo();
+
+        if (userInfo.getUserId() == null || Objects.equals(userInfo.getUserId(), userInfo.getVisitorId())) {
+            Log.d(TAG, "Invalid customer ID");
```

Comment:
sliiightly more precise message can have here. It's not 'invalid', it's 'not set'